force-app/main/default/classes/Trigger Recipes/MetadataTriggerHandler.cls

```diff
@@ -0,0 +1,131 @@
+/**
+ * @description This class exists as a unified, trigger handler class. It
+ * uses Custom Metadata, and introspection of the Trigger.new variable to
+ * determine what trigger handler classes should be called, and in what order.
+ *
+ * Metadata_Driven_Trigger__mdt has three fields:
+ *  * Object__c - is a metadata entity look up to an sObject ie: Account
+ *  * Execution_Order__c - is an integer and determines the order the trigger
+ *  *   handlers are executed
+ *  * Class__c - is a String holding the name of the Trigger Handler to execute
+ *
+ * Note: This Trigger framework works like this:
+ *
+ * An .trigger for a sObject invokes this class via:
+ * new MetadataTriggerHandler().run();
+ *
+ * This trigger handler class extends TriggerHandler - all the trigger handler
+ * classes _must_ extend trigger handler. Most classes will only overwrite the
+ * context methods like afterUpdate(). This class, however, overrides the run
+ * method. This class is responsible for determining which other trigger
+ * handler classes to instantiate and run.
+ *
+ * Concrete example:
+ * AccountTrigger.trigger (in this org) - invokes this class.
+ * This class queries the custom metadata and will find (at least) one metadata
+ * record tied to Account and the metadata record's Class__c specifies
+ * AccountTriggerHandler. This class then loops over the returned metadata
+ * records, instantiating the classes specified. It then calls the appropriate
+ * context methods on those classes.
+ *
+ */
+public with sharing class MetadataTriggerHandler extends TriggerHandler {
```

Comment:
`inherited sharing` would be better here.

Reply:
ugh. damn template. wonder when my patch for the template will be released.